coverage.py

```diff
@@ -89,7 +92,8 @@ def __init__(self, collection, uploader=None,
         # Since we are the metadata wrangler, any resources we find,
         # we mirror to S3.
         if not uploader:
-            uploader = S3Uploader()
+            uploader = S3Uploader.from_config(self._db)
```

Comment:
`Configuration.load()` calls `S3Uploader.integration()`, I think so we can store the bucket names at startup time and not have to keep checking them. 

But, it's true that we don't really need to store those bucket names at startup time. Unlike with CDN URLs we aren't checking the bucket names millions of times in a time-critical part of the code. 

So I think we could take out the reference to S3Uploader in Configuration.load without causing problems. Does that make sense?